### PR TITLE
[MIG][14.0] account_banking_ach_discount.

### PR DESCRIPTION
**Pull request by Murtaza-OSI**

### CODE REVIEW

**Comment by max3903** [comment]
@Murtaza-OSI Can you fix the checks issues?

File: account_banking_ach_discount/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                        for pay_li in mvl.bank_payment_line_id.payment_line_ids:
+                            # Get related payment line ref
+                            if pay_li.communication == inv_number:
+                                item["amount"] = pay_li.amount_currency
+                    # for non-ach payment
+                    # Deduct the discount only for the related payment.
+                    # Discount is applied on the last payment (i.e. fully reconciled).
+                    if (
+                        not mvl.bank_payment_line_id
+                        and mvl.move_id == self.id
```
Comment (atchuthan) [inline account_banking_ach_discount/models/account_move.py]: please change it to mvl.move_id.id == self.id

**Review by atchuthan — CHANGES_REQUESTED** [review]
We are getting warning in production due to this
Comparing apples and oranges: account.move(40585,) == 40606 (/odoo/external-src/l10n-usa/account_banking_ach_discount/models/account_move.py:41)

File: account_banking_ach_discount/models/account_move.py (hunk 1 above)
Comment (dreispt) [reply account_banking_ach_discount/models/account_move.py]: Even better, change to `mvl.move_id == self`

**Comment by dreispt** [comment]
@Murtaza-OSI plase fix pre-commit!

**Review by atchuthan — APPROVED** [review]

**Comment by dreispt** [comment]
This looks unfinished - precommit is failing

**Review by dreispt — APPROVED** [review]

**Pull request merged**